MapboxCoreNavigation/RouteController.swift

```diff
@@ -557,9 +581,14 @@ extension RouteController: CLLocationManagerDelegate {
         return isCloseToCurrentStep
     }
     
+    func resetPreviousDistanceArray() {
+        previousDistancesFromManeuver.removeAll()
```

Comment:
Is a separate method really needed for this statement? It looks like we’re calling it every time we touch `routeProgress.currentLegProgress.stepIndex` – maybe what this class needs is a property that mirrors `routeProgress.currentLegProgress.stepIndex`:

```swift
var currentStepIndex: Int {
    get {
        return routeProgress.currentLegProgress.stepIndex
    }
    set {
        routeProgress.currentLegProgress.stepIndex = newValue
        recentDistancesFromManeuver.removeAll()
    }
}
```

Reply:
As much as I like this style, I don't like hiding `routeProgress.currentLegProgress.stepIndex` behind another var. I like explicitly changing values on `routeProgress`.

Reply:
Two alternatives:

* Use KVO to observe for changes to the key path `routeProgress.currentLegProgress` and reset the array.
* Listen for alert level change notifications and reset the array.

Reply:
@1ec5 solved here: https://github.com/mapbox/mapbox-navigation-ios/pull/646/files#diff-1a2149692b3d7c58b6a0392120d6011aR566